code/game/objects/structures/aliens.dm

```diff
@@ -114,36 +252,53 @@
 	density = FALSE
 	layer = ABOVE_OPEN_TURF_LAYER
 	plane = FLOOR_PLANE
-	icon = 'icons/obj/smooth_structures/alien/weeds1.dmi'
-	icon_state = "weeds1"
-	base_icon_state = "weeds1"
+	icon = 'icons/obj/smooth_structures/alien/weeds.dmi'
+	icon_state = "weeds"
+	base_icon_state = "weeds"
 	max_integrity = 15
+	layer = 2.455
 	smoothing_flags = SMOOTH_BITMASK
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WEEDS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WEEDS, SMOOTH_GROUP_WALLS)
-	transform = matrix(1, 0, -4, 0, 1, -4)
 	var/obj/structure/alien/weeds/node/linked_node = null
 	var/static/list/weedImageCache
+	var/check_counter
 
-
-/obj/structure/alien/weeds/New(pos, node)
+/obj/structure/alien/weeds/Initialize(mapload, node)
 	..()
 	linked_node = node
 	if(isspaceturf(loc))
 		qdel(src)
 		return
-	if(icon_state == "weeds")
-		icon_state = pick("weeds", "weeds1", "weeds2")
-	spawn(rand(150, 200))
-		if(src)
-			Life()
+	for(var/turf/simulated/wall/W in orange(1, src))
+		var/wall_dir = get_dir(W, src)
+		switch(wall_dir)
+			if(1, 8, 4, 2)
+				new /obj/structure/alien/wallweed(loc, wall_dir)
+			if(9, 6, 5, 10)
+				continue
+	START_PROCESSING(SSobj, src)
 
 /obj/structure/alien/weeds/Destroy()
+	STOP_PROCESSING(SSobj, src)
 	QUEUE_SMOOTH_NEIGHBORS(src)
+	playsound(loc, pick('sound/effects/alien_resin_break2.ogg','sound/effects/alien_resin_break1.ogg'), 50, FALSE)
 	linked_node = null
 	return ..()
 
-/obj/structure/alien/weeds/proc/Life()
+/obj/structure/alien/weeds/attack_alien(mob/living/carbon/alien/humanoid/user)
+	if(user.a_intent != INTENT_HARM)
+		return
+	else
```

Comment:
no need for this else as above is a guard clause

Reply:
👍